packages/contract/src/contract-factory.ts

```diff
@@ -281,73 +294,90 @@ export default class ContractFactory {
-          result = await blobTx.waitForResult();
-        } catch (err: unknown) {
-          // Core will throw for blobs that have already been uploaded, but the blobId
-          // is still valid so we can use this for the loader contract
-          if ((<Error>err).message.indexOf(`BlobId is already taken ${blobId}`) > -1) {
-            // eslint-disable-next-line no-continue
-            continue;
+          let result: TransactionResult<TransactionType.Blob>;
+
+          try {
+            const blobTx = await account.sendTransaction(fundedBlobRequest);
+            result = await blobTx.waitForResult();
+          } catch (err: unknown) {
+            // Core will throw for blobs that have already been uploaded, but the blobId
+            // is still valid so we can use this for the loader contract
+            if ((<Error>err).message.indexOf(`BlobId is already taken ${blobId}`) > -1) {
+              // eslint-disable-next-line no-continue
+              continue;
+            }
+
+            throw new FuelError(ErrorCode.TRANSACTION_FAILED, 'Failed to deploy contract chunk');
           }
 
-          throw new FuelError(ErrorCode.TRANSACTION_FAILED, 'Failed to deploy contract chunk');
-        }
+          if (!result.status || result.status !== TransactionStatus.success) {
+            throw new FuelError(ErrorCode.TRANSACTION_FAILED, 'Failed to deploy contract chunk');
+          }
 
-        if (!result.status || result.status !== TransactionStatus.success) {
-          throw new FuelError(ErrorCode.TRANSACTION_FAILED, 'Failed to deploy contract chunk');
+          uploadedBlobs.push(blobId);
         }
-
-        uploadedBlobs.push(blobId);
       }
-    }
-
-    // Get the loader bytecode
-    const blobIds = chunks.map(({ blobId }) => blobId);
-    const loaderBytecode = getLoaderInstructions(blobIds);
 
-    // Deploy the loader contract via create tx
-    const { contractId, transactionRequest: createRequest } = this.createTransactionRequest({
-      bytecode: loaderBytecode,
-      ...deployOptions,
-    });
-    await this.fundTransactionRequest(createRequest, deployOptions);
-    const transactionResponse = await account.sendTransaction(createRequest);
-
-    const waitForResult = async () => {
+      await this.fundTransactionRequest(createRequest, deployOptions);
+      transactionId = createRequest.getTransactionId(account.provider.getChainId());
+      const transactionResponse = await account.sendTransaction(createRequest);
       const transactionResult = await transactionResponse.waitForResult<TransactionType.Create>();
       const contract = new Contract(contractId, this.interface, account) as TContract;
 
       return { contract, transactionResult };
     };
 
-    return { waitForResult, contractId, transactionId: transactionResponse.id };
+    const waitForTransactionId = async () => {
+      const maxPollingTime = 15_000_000; // 15 Minutes
```

Comment:
I believe that instead of doing it with polling you can to this:
```ts
let txIdResolver: (value: string | PromiseLike<string>) => void;
const txIdPromise = new Promise<string>((resolve) => {
  txIdResolver = resolve;
});
// ...

txIdResolver(createRequest.getTransactionId(account.provider.getChainId()))

// ...

return { waitForResult, contractId, waitForTransactionId: async () => txIdPromise }

```

Reply:
Way cleaner, thanks.

https://github.com/FuelLabs/fuels-ts/pull/2929/commits/980ec6675533ea5b32cfeabd72544e82ed01487c